backend/src/main/java/de/amos/apachepulsarui/service/ClusterService.java

```diff
@@ -16,17 +17,39 @@ public class ClusterService {
 
     private final PulsarAdmin pulsarAdmin;
 
-    public List<Cluster> getAllClusters() {
+    public List<ClusterDto> getAllClusters() {
         try {
             return pulsarAdmin.clusters().getClusters().stream()
-                    .map(cluster -> Cluster.builder()
-                            .id(cluster)
-                            .build())
+                    .map(ClusterDto::fromString)
+                    .map(this::enrichWithClusterData)
+                    .map(this::enrichWithBrokerData)
                     .toList();
         } catch (PulsarAdminException e) {
             log.error("Could not get list of all clusters. E: %s".formatted(e));
             return List.of();
         }
     }
 
+    private ClusterDto enrichWithClusterData(ClusterDto cluster) {
```

Comment:
Über das sollten wir auch nochmal sprechen, wie und wo bauen wir die Dtos zusammen. Geben wir alle Daten in das Dto und erzeugen direkt ein fertiges Objekt oder erzeugen wir das Dto im Service und reichern es dort mit daten an.